src/main/java/org/truffleruby/core/encoding/EncodingConverterNodes.java

```diff
@@ -479,8 +478,8 @@ protected Object setReplacement(RubyEncodingConverter encodingConverter, Object
         }
 
         @TruffleBoundary
-        private int setReplacement(EConv ec, byte[] string, int len, byte[] encodingName) {
-            return ec.setReplacement(string, 0, len, encodingName);
+        private int setReplacement(EConv ec, InternalByteArray byteArray, byte[] encodingName) {
+            return ec.setReplacement(byteArray.getArray(), byteArray.getOffset(), byteArray.getEnd(), encodingName);
```

Comment:
I think this should be `getLength()` for the 3rd arg given the parameter is `int len`.